### PR TITLE
IR-52: docker/distribution v2.7.1

### PR DESCRIPTION
**Pull request by dmage**
The image registry used some version of docker/distribution from master. This pull request updates docker/distribution to the release version 2.7.1.

Upstream changes:
  * ability to send error messages to bugsnag
  * [`autoredirect` option](https://docs.docker.com/registry/configuration/#token)

None of them are available/configurable in OCP.

### CODE REVIEW

**Comment by dmage** [comment]
/hold

This PR shouldn't be merged until it's signed off by QE.

**Comment by dmage** [comment]
go.sum conflict is fixed.

File: go.mod
Comment (dmage) [inline go.mod:45]: This will be replaced to github.com/openshift/docker-distribution after all approvals are there.

File: pkg/dockerregistry/server/auth.go
Comment (ricardomaraschini) [inline pkg/dockerregistry/server/auth.go:113]: This was made because an interface somewhere else has changed?

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

File: pkg/dockerregistry/server/auth.go
Comment (dmage) [reply pkg/dockerregistry/server/auth.go:113]: Yes, see vendor/github.com/docker/distribution/registry/auth/auth.go.

**Comment by dmage** [comment]
/assign @wzheng1 @bmcelvee @sferich888 

This PR got approval from the engineering team and is ready for next approvals. Please use `/label XXX-approved` (where XXX is `qe`, `docs`, or `px`) to approve this PR.

@wzheng1 - no new functionality is added there, so you may want to make regression testing.
@bmcelvee @sferich888 - no new functionality, should I ping you on such PRs so that you can ack no-doc/no-feature?

**Comment by wzheng1** [comment]
@dmage I have tested this pr, but the cluster I started up via cluster-bot is with version 4.6.0-0.ci.test-2020-10-20-063531-ci-ln-6g1cc4t, is it fine?

**Comment by dmage** [comment]
@wzheng1 it's a temporary version, I don't know if there is a way to know what PRs it contains. If you started your cluster using `launch openshift/image-registry#252`, then it is fine.

**Comment by wzheng1** [comment]
@dmage OK, get it, thanks! I launched a cluster with command: launch openshift/image-registry#252, and a cluster with version 4.6.0-0.ci.test-2020-10-20-063531-ci-ln-6g1cc4t is running for my test. Regression test with image registry is finished with no issue, so adding below label now:
/label qe-approved

**Comment by bmcelvee** [comment]
@dmage good question, if you think we should add no-doc items like this one to the 4.7 release notes, it couldn't hurt. Otherwise, it's probably not necessary. Regardless, I think this one is good.  

/label docs-approved

**Comment by dmage** [comment]
@ricardomaraschini I changed go.mod so that it uses the docker/distribution branch from the openshift org, not from my fork. Can you take a look and tag it again?

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

**Comment by dmage** [comment]
/hold
Something wrong with our CI.

**Pull request merged**
